src/main/webapp/profile.js

```diff
@@ -13,15 +13,21 @@
 // limitations under the License.
 
 function onProfileLoad() {
+  supplyLogoutLink();
   prepareFormValidation();
 }
 
-// Allows certain fields in the profile to be edited, hides edit button, and displays update button. 
-
-function onProfileLoad() {
-  supplyLogoutLink();
+// Submits profile form to Datastore.
+function submitProfileForm(methodType, redirectUrl) {
+  fetch(`/person?user-email=${$("#user-email").val()}&first-name=${$("#first-name-field").val()}`+
+    `&last-name=${$("#last-name-field").val()}&company=${$("#company-field").val()}&job=`+
+    `${$("#job-field").val()}&linkedin=${$("#linkedin-field").val()}`, 
+    {method: methodType})
+    .then(window.location.replace(redirectUrl));
```

Comment:
Is there a strong reason you prefer having this URL-encoded? If not, it should be JSON text in the request body (see Meagan's availability update code for an example). The "user" JSON object should have a Java equivalent class.

Reply:
Changed everything to use the request body strategy.